airflow/utils/helpers.py

```diff
@@ -305,6 +305,9 @@ def on_terminate(p):
     except OSError as err:
         if err.errno == errno.ESRCH:
             return
+        # If operation not permitted error is thrown due to run_as_user, use sudo to kill the process
+        if err.errno == errno.EPERM:
+            os.system('sudo kill -' + str(sig) + ' ' + str(os.getpgid(pid)))
```

Comment:
Two things:

1. Add `−-non-interactive` to the command line - we don't ever want to prompt for a password (as there's no user to give it)
2. Instead of using `os.system` and passing it a string we should use subprocess with an array: https://docs.python.org/2/library/subprocess.html#replacing-bin-sh-shell-backquote

Reply:
I have added --non-interactive option and also used subprocess to execute the command